src/libostree/ostree-repo.c

```diff
@@ -3184,8 +3184,13 @@ reload_sysroot_config (OstreeRepo          *self,
      * binary "x" in /usr/lib/ostree/bootloaders/x). See:
      * https://github.com/ostreedev/ostree/issues/1719
      * https://github.com/ostreedev/ostree/issues/1801
+     * Also, dedup these strings with the bootloader implementations
      */
-    if (!(g_str_equal (bootloader, "auto") || g_str_equal (bootloader, "none")))
+    if (!(g_str_equal (bootloader, "auto") || g_str_equal (bootloader, "none")
+          || g_str_equal (bootloader, "grub2")
+          || g_str_equal (bootloader, "syslinux")
+          || g_str_equal (bootloader, "U-Boot")
```

Comment:
If we allow these other ones too now, we should also handle them in `ostree_sysroot_write_deployments_with_options`, no?

Reply:
Probably...it just grows into a bigger PR that has the potential to break other things and I'd like to fast track zipl support into the next release and circle back and clean up the bootloader bits after that.  Sound OK?

Reply:
Per IRC chat, removed the non-zipl changes here.